pkg/openstack/go.mod

```diff
@@ -0,0 +1,61 @@
+module github.com/openstack-k8s-operators/lib-common/pkg/openstack
```

Comment:
this creates a nested module and gopls complains about nested modules in vscode. Would it be possible to not introduce nested modules (directories with go.mod in it)?

Reply:
like its mentioned there, you can fix/workaround this buy open each module in its own workspace folder (File -> Add Folder to Workspace). There is currently the plan to introduce sub modules in the operators themselves to have dedicated module for the apis as this will reduce the go.mod dependencies on operators which just use other operators types.

Reply:
that workaround only helps to have a second instance of that directory in the workspace tree, and browse it without gopls errors. But when you go into the main repo directory (lib-common) with subpackages in it, it explodes with "problems". I would suggest to not use submodules because of that.

Reply:
with https://github.com/openstack-k8s-operators/lib-common/pull/29 we restructure lib-common so that it is easier to work with and only have modules in /modules. you'd then still need to open each module, but the main repo dir can basically ignored, if there is no need to add a new module.